### PR TITLE
add Lock bindings to thread.c

### PR DESCRIPTION
**Pull request by ConstNW**
porting nekovm code by @ncannasse  [suggestion](https://github.com/HaxeFoundation/haxe/issues/8699#issuecomment-525023553) at HaxeFoundation/haxe/issues/8699

### CODE REVIEW

File: src/std/thread.c [hunk 1]
```diff
@@ -338,6 +340,112 @@ DEFINE_PRIM(_VOID, deque_add, _DEQUE _DYN);
...
 
+// ----------------- LOCK
+
+struct _hl_lock;
+typedef struct _hl_lock hl_lock;
+
+struct _hl_lock {
+	void (*free)( hl_lock * );
+#if !defined(HL_THREADS)
+	void *value;
```
Comment (ncannasse) [inline src/std/thread.c]: why having an unused value here?

Comment (ConstNW) [reply src/std/thread.c]: it's useless.

**Comment by Simn** [comment]
Can we merge this? Quick reply appreciated!

**Comment by sonygod** [comment]
@Simn it's work!

**Pull request merged**